server/webhook.go

```diff
@@ -21,9 +20,18 @@ const (
 )
 
 func (p *Plugin) handleWebhook(w http.ResponseWriter, r *http.Request) {
-	params := mux.Vars(r)
-
-	if subtle.ConstantTimeCompare([]byte(p.configuration.WebhookSecret), []byte(params["webhooksecret"])) == 0 {
+	secret := r.URL.Query().Get("secret")
+	config := p.getConfiguration()
+	if err := config.IsValid(); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if config.WebhookSecret == "" {
+		http.Error(w, "Webhook not found in config", http.StatusInternalServerError)
```

Comment:
We don't want to reveal via API too many details about the situation of missing the secret

```suggestion
              http.Error(w, "Not authorized", http.StatusUnauthorized)
```

Reply:
Done